rfcs/text/0015_nodejs_clustering.md

```diff
@@ -0,0 +1,641 @@
+
+When using `cluster`, the common best practice is to have the coordinator recreate ('restart') workers when they terminate unexpectedly. 
+However, given Kibana's architecture, some failures are not recoverable (workers failing because of config validation, failed migration...). 
+
+For instance, if a worker (well, all workers) terminates because of an invalid configuration property, it doesn't make
+any sense to have the coordinator recreate them indefinitely, as the error requires manual intervention. 
+
+As a first step, we plan to terminate the main Kibana process when any worker terminates unexpectedly for any reason (after all,
+this is already the behavior in non-cluster mode). In the future, we will look toward distinguishing between recoverable
+and non-recoverable errors as an enhancement, so that we can automatically restart workers on any recoverable error.
+
+### 6.1.8 Data folder
+
+The data folder (`path.data`) is currently the same for all workers. 
+
+We still have to identify with the teams if this is going to be a problem. It could be, for example, if some plugins
+are accessing files in write mode, which could result in concurrency issues between the workers.
+
+If that was confirmed, we would plan to create and use a distinct data folder for each worker, which would be non-breaking
+as we don't consider the layout of this directory to be part of our public API.
+
+### 6.1.9 instanceUUID
+
+The same instance UUID (`server.uuid` / `{dataFolder}/uuid`) is currently used by all the workers. 
+
+So far, we have not identified any places where this will be problematic, however, we will look to other teams to
+help validate this.
+
+Note that if we did need to have per-worker UUIDs, this could be a breaking change, as the single `server.uuid`
+configuration property would not be enough. If this change becomes necessary, one approach could be to have unique worker
+IDs with `${serverUuid}-${workerId}`.
+
+## 6.2 Technical impact on Plugins
+
+### 6.2.1 What types of things could break?
+
+#### Concurrent access to the same resources
+
+Is there, for example, some part of the code that is accessing and writing files from the data folder (or anywhere else) 
+and makes the assumption that it is the sole process actually writing to that file?
+
+#### Using instanceUUID as a unique Kibana process identifier
+
+Is there, for example, schedulers that are using the instanceUUID a single process id, in opposition to a single 
+Kibana instance id? Are there situations where having the same instance UUID for all the workers is going to be a problem?
+
+#### Things needing to run only once per Kibana instance
+
+Is there any part of the code that needs to be executed only once in a multi-worker mode, such as initialization code, 
+or starting schedulers? 
+
+An example would be Reporting's queueFactory pooling. As we want to only be running a single headless at a time per 
+Kibana instance, only one worker should have pooling enabled.
+
```

Comment:
> we want to only be running a single headless at a time per Kibana instance

If we didn't do anything, the number of Chromium processes running would always be <= to the number of cores. That seems OK to me.